api/v1/imagescan/service.go

```diff
@@ -177,18 +170,21 @@ func (s *serviceImpl) GetImageComponents(ctx context.Context, req *v1.GetImageCo
 	}, nil
 }
 
-func (s *serviceImpl) getImageComponents(ctx context.Context, req *v1.GetImageComponentsRequest, uncertifiedRHEL bool) (*apiV1.ComponentsEnvelope, error) {
-	imageScan, err := s.ScanImage(ctx, &v1.ScanImageRequest{
-		Image:           req.GetImage(),
-		Registry:        req.GetRegistry(),
-		UncertifiedRHEL: uncertifiedRHEL,
-	})
+func (s *serviceImpl) getImageComponents(_ context.Context, req *v1.GetImageComponentsRequest, uncertifiedRHEL bool) (*apiV1.ComponentsEnvelope, error) {
+	image, _, err := s.processImage(req.GetImage(), req.GetRegistry(), uncertifiedRHEL)
 	if err != nil {
 		return nil, err
 	}
 
+	// If a digest was provided as part of the request, that same digest must be used in the call to getLayer.
+	// A different digest will result in a NotFound error (observed when the request digest represents a ManifestList)
+	//
+	// server.processImage will ensure image.SHA is populated with the digest from request if exists, otherwise will be first layer digest
```

Comment:
I'm confused, how do we guarantee `image.SHA` is populated?

Reply:
Here: https://github.com/stackrox/scanner/blob/d1c1e03ddb63e367a45d47dbe462b316bcbc15da/pkg/scan/scan.go#L75-L77